tests/functional/base.py

```diff
@@ -17,8 +20,105 @@
 from custodia.server.config import parse_config
 
 
+def wait_pid(process, wait):
+    timeout = time.time() + wait
+    while time.time() < timeout:
+        pid, _ = os.waitpid(process.pid, os.WNOHANG)
+        if pid == process.pid:
+            return True
+        time.sleep(0.1)
+    return False
+
+
+def wait_socket(process, custodia_socket, wait):
+    timeout = time.time() + wait
+    while time.time() < timeout:
+        if process.poll() is not None:
+            raise AssertionError(
+                "Premature termination of Custodia server")
+        try:
+            s = socket.socket(family=socket.AF_UNIX)
+            s.connect(custodia_socket)
+        except OSError:
+            pass
+        else:
+            return True
+        time.sleep(0.1)
+    raise OSError('Timeout error')
+
+
+def translate_meta_uid(meta_uid):
+    current_uid = None
+
+    if meta_uid == "correct_id":
+        current_uid = pwd.getpwuid(os.geteuid()).pw_uid
```

Comment:
This is kinda pointless, ``SOCK_PEERCRED`` always returns the effective uid and gid.